airbyte-integrations/connectors/source-slack/source_slack/spec.json

```diff
@@ -0,0 +1,144 @@
+      },
+      "channel_filter": {
+        "type": "array",
+        "default": [],
+        "items": {
+          "type": "string",
+          "minLength": 0
+        },
+        "title": "Channel name filter",
+        "description": "A channel name list (without leading '#' char) which limit the channels from which you'd like to sync. Empty list means no filter.",
+        "examples": ["channel_one", "channel_two"]
+      },
+      "credentials": {
+        "title": "Authentication mechanism",
+        "description": "Choose how to authenticate into Slack",
+        "type": "object",
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "Sign in via Slack (OAuth)",
+            "required": [
+              "option_title",
+              "client_id",
+              "client_secret",
+              "access_token"
+            ],
+            "properties": {
+              "option_title": {
+                "type": "string",
+                "const": "Default OAuth2.0 authorization"
+              },
+              "client_id": {
+                "type": "string",
+                "title": "Client ID",
+                "description": "Slack client_id. See our <a href=\"https://docs.airbyte.com/integrations/sources/slack\">docs</a> if you need help finding this id."
+              },
+              "client_secret": {
+                "type": "string",
+                "title": "Client Secret",
+                "description": "Slack client_secret. See our <a href=\"https://docs.airbyte.com/integrations/sources/slack\">docs</a> if you need help finding this secret.",
+                "airbyte_secret": true
+              },
+              "access_token": {
+                "type": "string",
+                "title": "Access token",
+                "description": "Slack access_token. See our <a href=\"https://docs.airbyte.com/integrations/sources/slack\">docs</a> if you need help generating the token.",
+                "airbyte_secret": true
+              }
+            },
+            "order": 0
+          },
+          {
+            "type": "object",
+            "title": "API Token",
+            "required": ["option_title", "api_token"],
+            "properties": {
+              "option_title": {
+                "type": "string",
+                "const": "API Token Credentials"
+              },
+              "api_token": {
+                "type": "string",
+                "title": "API Token",
+                "description": "A Slack bot token. See the <a href=\"https://docs.airbyte.com/integrations/sources/slack\">docs</a> for instructions on how to generate it.",
+                "airbyte_secret": true
+              }
+            },
+            "order": 1
+          }
+        ]
+      }
+    }
+  },
+  "advanced_auth": {
```

Comment:
Since this connector supports OAuth2, we need to provide an `oauth2.patch.json` as well to specify how the OAuth2 flow for this connector would work. To do that, you might want to look at Slack's documentation on OAuth2 to see how it works for them. The airbyte specification in this file can also be useful although might be a bit cryptic to read.